src/components/ColorChoice.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+
+const ColorChoice = (props) => {
+  const [color, setColor] = useState('#000000'); 
+
+  const handleColorChange = (e) => {
+    const selectedColor = e.target.value;
+    setColor(selectedColor);
+    props.setColorCallback(selectedColor);
+  };
+
+  return (
+    <div className="color-choice">
+      <h3>Select Color:</h3>
+      <input
+        type="color"
+        value={color}
+        onChange={handleColorChange}
+      />
```

Comment:
Awesome exploration of input types! If we wanted to take this further, what would we need to change about the project if we wanted the user to be able to select different colors for each participant in the chat?